### ISSUE TITLE
[Feature request] Mouse support on TTY

### ISSUE THREAD

**Issue by magiblot** [asciinema/asciinema#333]
Running asciinema on a X11 terminal doesn't prevent mouse-compatible programs from interacting with the mouse. However, programs which support GPM on TTY can't use the mouse when being recorded with asciinema.

I guess the mouse pointer would have

**Comment by Screwtapello** [comment]
I'm guessing this would be very difficult, or impossible.

asciinema works by putting itself between the terminal it's running in, and the program being run, and capturing all the communication between them. The xterm mouse protocol is part of that communication stream, so it naturally gets captured along with everything else. However, the GPM mouse protocol is *not* part of that communication stream - `gpm` is a separate program, unrelated to the terminal, and programs talk to it via very different means that asciinema knows nothing about.

**Comment by magiblot** [comment]
Yeah, you are right. It would be possible to translate GPM signals into xterm ones, like https://github.com/cosmos72/twin does, but there is no point in making asciinema this complex. Another way would be letting GPM signals through asciinema somehow so that underlying applications can interact directly with it, but I have never seen an application able to do this, so I'm not sure this is even possible.

I guess recording the GPM pointer would be impossible in any case, although some applications hide it and draw their own pointer instead (which is what I think twin does).

**Comment by magiblot** [comment]
Okay, I know how to do this:

One way is to pass the stdin file descriptor directly to the process using GPM, because it's a console file descriptor.

Another way (probably simplier) is to let asciinema work in `tee` mode. `tee` gets some input from stdin and dumps it into a file while printing it again in stdout, but it doesn't include timing information. So ideally I could record my application by running `./app | asciinema rec --from-stdin` and asciinema would exit automatically after the pipe breaks.

**Comment by magiblot** [comment]
A possible workaround is to create a named pipe (e.g. `mknod pipe p`), start asciinema and do `cat pipe`. Then, from a different tty, run `./app | tee pipe`.
